aws-rds-cfn-common/src/main/java/software/amazon/rds/common/logging/RequestLogger.java

```diff
@@ -1,109 +1,123 @@
 
 import org.apache.commons.lang3.ObjectUtils;
 import org.apache.commons.lang3.StringUtils;
 import org.apache.commons.lang3.exception.ExceptionUtils;
-import org.json.JSONObject;
 
-import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
 import lombok.NonNull;
+import software.amazon.awssdk.awscore.AwsRequest;
+import software.amazon.awssdk.awscore.AwsResponse;
+import software.amazon.awssdk.core.ResponseBytes;
+import software.amazon.awssdk.core.ResponseInputStream;
 import software.amazon.cloudformation.proxy.Logger;
 import software.amazon.cloudformation.proxy.ProgressEvent;
-import software.amazon.cloudformation.proxy.ProxyClient;
 import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
 import software.amazon.rds.common.printer.JsonPrinter;
 
 @lombok.Getter
 @lombok.Setter
 public class RequestLogger {
 
-    public static final String MESSAGE_MARKER = "Message";
+    public static final String MESSAGE_EVENT = "Message";
     public static final String THROWABLE_MARKER = "Throwable";
-    public static final String REQUEST_DATA_MARKER = "RequestData";
+    public static final String MESSAGE = "Message";
+    public static final String STACK_ID = "StackId";
+    public static final String AWS_ACCOUNT_ID = "AwsAccountId";
+    public static final String CLIENT_REQUEST_TOKEN = "ClientRequestToken";
     private final Logger logger;
-    private final JSONObject requestData;
+    private final Map<String, String> requestDataMap;
     private final JsonPrinter jsonPrinter;
 
+    private final LogRuleSet DEFAULT_LOG_RULE_SET = LogRuleSet.builder()
+            .withLogClasses((Throwable t) -> log(t),
+                    Throwable.class)
+            .withLogClasses((Iterable<?> it) -> it.forEach(r -> log(r)),
+                    Iterable.class)
+            .withLogClasses((ResponseBytes<?> rb) -> log(rb.response().getClass().getSimpleName(), rb.response()),
+                    ResponseBytes.class)
+            .withLogClasses((ResponseInputStream<?> is) -> log(is.response().getClass().getSimpleName(), is.response()),
+                    ResponseInputStream.class)
+            .withLogClasses((String s) -> log(s, s),
+                    String.class)
+            .withLogClasses((Object r) -> log(r.getClass().getSimpleName(), r),
+                    AwsRequest.class,
+                    AwsResponse.class,
+                    CompletableFuture.class,
+                    Object.class)
+            .build();
+
     public <T> RequestLogger(final Logger logger,
                              final @NonNull ResourceHandlerRequest<T> request,
                              final JsonPrinter jsonPrinter) {
         this.logger = logger;
         this.jsonPrinter = jsonPrinter;
-        requestData = new RequestData(request.getStackId(), request.getAwsAccountId(), request.getClientRequestToken())
-                .toJson();
+        this.requestDataMap = Maps.newHashMap();
```

Comment:
NIT: if I am not mistaken,`new HashMap<>();` would eliminate the new import. As we do not explicitly import the underlying library, it might be better to use the stdlib instead.

Reply:
Done.